outputs.tf

```diff
@@ -35,12 +48,12 @@ output "security_group_ids" {
 
 output "role" {
   description = "Name of AWS IAM Role associated with creating instance"
-  value       = "${join("", aws_iam_role.default.*.name)}"
+  value       = "${join(",", aws_iam_role.default.*.name)}"
```

Comment:
why `,`?
the `splat+join` pattern uses `join(""` and will create a single string from an attribute of the resource with `count`.

Reply:
Hi @aknysh 
I'll show you :)

If `count` is `0`, then `"${join("", aws_iam_role.default.*.name)}"` returns `""`
If count is `1`, then `"${join("", aws_iam_role.default.*.name)}"` returns `"some_role_name_1"`
If count is `2`, then `"${join("", aws_iam_role.default.*.name)}"` returns `"some_role_name_1some_role_name_2"`

If `count` is `0`, then `"${join(",", aws_iam_role.default.*.name)}"` returns `""`
If count is `1`, then `"${join(",", aws_iam_role.default.*.name)}"` returns `"some_role_name_1"`
If count is `2`, then `"${join(",", aws_iam_role.default.*.name)}"` returns `"some_role_name_1, some_role_name_2"`

If `count` is `0`, then `"${aws_iam_role.default.*.name}"` returns `Some error`
If count is `1`, then `"${aws_iam_role.default.*.name}"` returns `["some_role_name_1"]`
If count is `2`, then `"${aws_iam_role.default.*.name}"` returns `["some_role_name_1, some_role_name_2"]`

Because the output of `join()` is a string, i was keeping with what your previous output was.
And now that it can do multiple servers, I have it separated by a comma, which is easy enough to `split()` as an input.

Does that clarify enough?